symengine/tests/basic/test_sets.cpp

```diff
@@ -690,16 +698,22 @@ TEST_CASE("ConditionSet : Basic", "[basic]")
     REQUIRE(eq(*r1->contains(integer(3)), *boolTrue));
     REQUIRE(eq(*r1->contains(integer(-3)), *boolFalse));
     REQUIRE(eq(*r1->contains(integer(2)), *boolFalse));
-    REQUIRE(eq(*r1->contains(finiteset({integer(3)})), *boolTrue));
+    REQUIRE(eq(*r1->contains(finiteset({integer(3)})), *boolFalse));
```

Comment:
@isuruf, this test case is failing as  `And(9 == {3}**2, Contains({3}, (1, oo)))` doesn't evaluate to `boolFalse`.
Is this a problem with `Interval:: contains()`?